### ISSUE TITLE
Search in the documentation?

### ISSUE THREAD

**Issue by macintacos** [federico-terzi/espanso#428]
I'm porting a lot of stuff that I have over from Typinator to Espanso (for a variety of reasons, one of which being that Espanso's text-base config is 👨‍🍳💋) but one thing that has bothered me so far is that there isn't search in [the documentation](https://espanso.org/docs/). Any chance that could be added?

**Comment by federico-terzi** [comment]
Hey @macintacos,
Thanks for the kind words :) Yes, that could be a useful feature to have. I'll add it to the roadmap!

Cheers :)

**Comment by AmberHaelo** [comment]
Meanwhile, as with most websites that do not prohibit spiders, you can use popular search engines to trawl even specific directories of sites, without any special technology being added to every site in order to do so—and often search engines will have a better index and search syntax than local site code anyway. For example, here we search for "[advanced forms](https://duckduckgo.com/?q=advanced+forms+site%3Aespanso.org%2Fdocs)". And if you've got a browser that lets you do it, you can save a bookmark/custom search with keyword and a wildcard, then access the search with something like "edoc *term*" in the URL field. Here is how you would set up Vivaldi in such a way:

![](http://www.semigreenpenny.com/img/forums/20244521-espanso-site_specific_search.png)

I've got dozens of these! And of course if you are using a simpler browser, you can use Espanso to help create the search URLs for you.

**Comment by federico-terzi** [comment]
@AmberHaelo Is there a URL format to a Duckduckgo search over a specific website? That way we could create an input box on the top of the documentation redirecting to the Duckduckgo search results

**Comment by AmberHaelo** [comment]
Yeah, they have a [kit page](https://duckduckgo.com/search_box) for putting together a search tool for embedding in your site.

**Comment by federico-terzi** [comment]
@AmberHaelo Great! Thank you for pointing this out

**Comment by chrisgrieser** [comment]
I came here to mention this as well! a search function would be really useful.

I just started using because the regex triggers are just too enticing and getting everything to work properly took me a bit without having a proper search. Nevertheless, fantastic application, the configurability is fantastic.
